platform.txt

```diff
@@ -198,22 +198,22 @@ tools.openocd.cmd.windows=bin/openocd.exe
 
 tools.openocd.upload.params.verbose=-d2
 tools.openocd.upload.params.quiet=-d0
-tools.openocd.upload.pattern="{path}/{cmd}" {upload.verbose} -s "{path}/share/openocd/scripts/" -f "{runtime.platform.path}/variants/{build.variant}/{build.openocdscript}" -c "telnet_port disabled; program {{build.path}/{build.project_name}.bin} verify reset 0x2000; shutdown"
+tools.openocd.upload.pattern="{path}/{cmd}" {upload.verbose} -s "{path}/share/openocd/scripts/" -f "interface/{protocol}" -c "set telnet_port 0" {extra_params} -f "target/at91samdXX.cfg" -c "telnet_port disabled; program {{build.path}/{build.project_name}.bin} verify reset 0x2000; shutdown"
 
 tools.openocd.network_cmd={runtime.tools.arduinoOTA.path}/bin/arduinoOTA
 tools.openocd.upload.network_pattern={network_cmd} -address {serial.port} -port 65280 -username arduino -password "{network.password}" -sketch "{build.path}/{build.project_name}.bin" -upload /sketch -b
 
 tools.openocd.program.params.verbose=-d2
 tools.openocd.program.params.quiet=-d0
-tools.openocd.program.pattern="{path}/{cmd}" {program.verbose} -s "{path}/share/openocd/scripts/" -f "{runtime.platform.path}/variants/{build.variant}/{build.openocdscript}" -c "telnet_port disabled; program {{build.path}/{build.project_name}.hex} verify reset; shutdown"
+tools.openocd.program.pattern="{path}/{cmd}" {program.verbose} -s "{path}/share/openocd/scripts/" -f "interface/{protocol}" -c "set telnet_port 0" {extra_params} -f "target/at91samdXX.cfg" -c "telnet_port disabled; program {{build.path}/{build.project_name}.hex} verify reset; shutdown"
```

Comment:
What's the deal with `telnet_port disabled` and `set telnet_port 0` here? Seems duplicate?

Reply:
`set telnet_port 0` was hidden in the variant configuration; it was necessary alongside `telnet_port disabled` due to this issue https://github.com/arduino/Arduino/issues/4330 , so I'd leave it there.

Reply:
Hm, I'm a bit confused about that issue you linked. Sounds like that issue is really only about gdb_port not accepting `disabled` (but that was since fixed in openocd upstream, not sure if the Arduino version already updated to include that fix)? And it seems that `disabled` is the right value for all openocd versions (using `0` would, AFAICS, in older version be used as disabled, but in newer versions cause a random port to be used).

So are you *sure* that `set telnet_port 0` is needed? I would think it never was really required, even?

Also, I can imagine that `tcl_port disabled` and (if openocd is sufficiently new) `gdb_port disabled` would make sense here as well? Or are we actually using the gdb_port over TCP now (as opposed to using it over stdin/stdout)?

Reply:
Not _sure_ it's still needed, the best way to check this would be trying to reproduce the issue but it's a bit impractical. About `gdb_port disabled`, I need to check how the debugging bridge is implemented in IDE 2.0 ; is it's over stdio we can safely add it.